lib/Dialect/Calyx/Transforms/CalyxLoweringUtils.cpp

```diff
@@ -843,5 +843,88 @@ BuildCallInstance::getCallComponent(mlir::func::CallOp callOp) const {
   return nullptr;
 }
 
+PredicateInfo getPredicateInfo(CmpFPredicate pred) {
+  using CombLogic = PredicateInfo::CombLogic;
+  using Port = PredicateInfo::InputPorts::Port;
+  PredicateInfo info;
+  switch (pred) {
+  case CmpFPredicate::OEQ: {
+    return PredicateInfo{.logic = CombLogic::And,
+                         .inputPorts = {{Port::Eq, /*inverted=*/false},
+                                        {Port::Unordered, /*inverted=*/true}}};
```

Comment:
This designated initializer is c++ 20 feature so can you replace it with plain one?